warehouse/query-core/src/main/java/datawave/query/iterator/builder/IvaratorBuilder.java

```diff
@@ -1,35 +1,65 @@
 package datawave.query.iterator.builder;
 
+import datawave.query.iterator.ivarator.IvaratorCacheDir;
 import datawave.core.iterators.querylock.QueryLock;
 import datawave.query.composite.CompositeMetadata;
 import datawave.query.iterator.profile.QuerySpanCollector;
 import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
 
 /**
  * A base class used to build ivarators
  */
 public abstract class IvaratorBuilder extends IndexIteratorBuilder {
     
-    protected FileSystem hdfsFileSystem;
+    protected List<IvaratorCacheDir> ivaratorCacheDirs;
     protected String hdfsFileCompressionCodec;
     protected QueryLock queryLock;
-    protected String ivaratorCacheDirURI;
     protected long ivaratorCacheScanPersistThreshold = 100000L;
     protected long ivaratorCacheScanTimeout = 1000L * 60 * 60;
     protected int ivaratorCacheBufferSize = 10000;
     protected int maxRangeSplit = 11;
     protected int ivaratorMaxOpenFiles = 100;
+    protected int ivaratorNumRetries = 2;
     protected boolean collectTimingDetails = false;
     protected QuerySpanCollector querySpanCollector = null;
     protected CompositeMetadata compositeMetadata;
     protected int compositeSeekThreshold;
     
-    public FileSystem getHdfsFileSystem() {
-        return hdfsFileSystem;
+    protected void validateIvaratorCacheDirs(List<IvaratorCacheDir> ivaratorCacheDirs) {
+        if (ivaratorCacheDirs.isEmpty())
+            throw new IllegalStateException("No ivarator cache dirs defined");
+        
+        IvaratorCacheDir ivaratorCacheDir = ivaratorCacheDirs.get(0);
```

Comment:
I think originally I had the IvaratorCacheDirConfig objects being validated here using their isValid() method, but I ended up moving that into the IteratorBuildingVisitor instead.  So, by this point, the ivarator cache dir configs have already been validated.  This function is just making sure that the control directory exists, since we know that path/fs will be needed by the ivarator.

Reply:
updated.